src/traversals/bfs.jl

```diff
@@ -58,46 +58,48 @@ bfs_tree(g::AbstractGraph, s::Integer; dir=:out) = tree(bfs_parents(g, s; dir=di
     gdistances!(g, source, dists)
 
 Fill `dists` with the geodesic distances of vertices in `g` from `source`.
-`dists` should be a vector of length `nv(g)`. Return `dists`.
-For vertices in disconnected components the default distance is -1.
```

Comment:
Is this for unsigned integer graph types? What about the edge case where `nv == typemax` and `g` is a path graph for example? This is more likely with smaller int types.

Reply:
if `nv == typemax` then you're going to have massive problems across the board. The bottom line is that there has to be some upper sentinel. If you've got a 256-node graph and you're trying to shoehorn it into a `UInt8`, you're gonna have a bad time.

Reply:
(In fact, we should probably ensure that `squash` doesn't do that.)